File: Sorting/bubble.py
def bubble_sort(arr):
    for i in range(len(arr)-1):
        for j in range(len(arr)-1-i):
            if arr[j] > arr[j+1]:
                temp = arr[j]
                arr[j] = arr[j+1]
                arr[j+1] = temp
    return arr


if __name__ == "__main__":
    arr = [10, 8, 5, 6, 4, 1, 2, 3, 7, 9]

    print(arr)
    print(bubble_sort(arr))

File: Sorting/bubble.ts
enum SortDirection {
  Asc = "asc",
  Desc = "dsc",
}

const bubbleSort = (arr: number[], direction: SortDirection): number[] => {
  const ans: number[] = arr.map((element) => element);
  for (let index = 0; index < ans.length; index++) {
    for (let indexTwo = 0; indexTwo < ans.length - index; indexTwo++) {
      const firstElement: number = ans[indexTwo];
      const secondElement: number = ans[indexTwo + 1];

      const condition: boolean =
        direction === SortDirection.Asc
          ? firstElement > secondElement
          : secondElement > firstElement;

      if (condition) {
        ans[indexTwo] = secondElement;
        ans[indexTwo + 1] = firstElement;
      }
    }
  }
  return ans;
};

const arr: number[] = [10, 8, 5, 6, 4, 1, 2, 3, 7, 9];
const ansAsc: number[] = bubbleSort(arr, SortDirection.Asc);
const ansDsc: number[] = bubbleSort(arr, SortDirection.Desc);

console.log(arr, ansAsc, ansDsc);
